File: tests/fixtures/cpp/classes.cpp
#include <string>

class Animal {
public:
    Animal(std::string name) : name_(name) {}
    virtual ~Animal() = default;

    virtual std::string speak() { return "..."; }
    std::string get_name() { return name_; }

private:
    std::string name_;
};

class Dog : public Animal {
public:
    Dog(std::string name) : Animal(name) {}

    std::string speak() override { return "Woof!"; }
};

File: tests/fixtures/javascript/classes.js
class Animal {
    constructor(name) {
        this.name = name;
    }

    speak() {
        return "Some sound";
    }

    async move() {
        console.log("Moving...");
    }
}

class Dog extends Animal {
    speak() {
        return "Woof!";
    }
}

export class Service {
    process(data) {
        return data;
    }
}
